src/ILLink.RoslynAnalyzer/DataFlow/LocalDataFlowVisitor.cs

```diff
@@ -221,5 +201,49 @@ public override TValue VisitConversion (IConversionOperation operation, LocalDat
 			var operandValue = Visit (operation.Operand, state);
 			return operation.OperatorMethod == null ? operandValue : TopValue;
 		}
+
+		public override TValue VisitObjectCreation (IObjectCreationOperation operation, LocalDataFlowState<TValue, TValueLattice> state)
+		{
+			if (operation.Constructor == null)
```

Comment:
Just curious, do you know if we can actually get here without a ctor?

Reply:
I didn't try - but I would expect this to happen for compiler generated .ctors - as those don't exist yet. So for example class without explicit constructor - it very likely doesn't have any method symbol for the default .ctor, but it will eventually get one.

Fortunately we don't need to check for that in data flow. RUC needs to do that though and check for RUC on type in that case.